### PR TITLE
Reading of optics data

### PR DESCRIPTION
**Pull request by jmmshn**
## Added LOPTICS output parsing
- Modified the drone reading of dielectric function and absorption spectrum
- Added tests

### CODE REVIEW

File: atomate/vasp/tests/test_drones.py [hunk 1]
```diff
@@ -53,124 +63,168 @@ def test_runs_assimilate(self):
...
         self.assertIsNotNone(pot_spec[0]["hash"])  # check a hash was loaded
 
         # Force not loading of POTCAR
         drone = VaspDrone(parse_potcar_file=False)
         doc = drone.assimilate(self.Si_static)
-        pot_spec = doc['calcs_reversed'][0]['input']["potcar_spec"]
+        pot_spec = doc["calcs_reversed"][0]["input"]["potcar_spec"]
         self.assertIsNone(pot_spec[0]["hash"])  # check a hash was not loaded
+
+    def test_parse_optical(self):
```
Comment (jmmshn) [inline atomate/vasp/tests/test_drones.py:217]: New test is here

File: atomate/vasp/drones.py [hunk 2]
```diff
@@ -449,10 +575,20 @@ def process_vasprun(self, dir_name, taskname, filename):
 
         # perform Bader analysis using Henkelman bader
         if self.parse_bader and "chgcar" in d["output_file_paths"]:
-            suffix = '' if taskname == 'standard' else ".{}".format(taskname)
+            suffix = "" if taskname == "standard" else ".{}".format(taskname)
             bader = bader_analysis_from_path(dir_name, suffix=suffix)
             d["bader"] = bader
 
+        # parse output from loptics
```
Comment (jmmshn) [inline atomate/vasp/drones.py:582]: check vrun for the optical data

File: atomate/vasp/drones.py [hunk 3]
```diff
@@ -300,31 +384,47 @@ def generate_doc(self, dir_name, vasprun_files, outcar_files):
...
             if d["input"]["parameters"].get("LEPSILON"):
-                for k in ['epsilon_static', 'epsilon_static_wolfe', 'epsilon_ionic']:
+                for k in ["epsilon_static", "epsilon_static_wolfe", "epsilon_ionic"]:
                     d["output"][k] = d_calc_final["output"][k]
                 if SymmOp.inversion() not in sg.get_symmetry_operations():
                     for k in ["piezo_ionic_tensor", "piezo_tensor"]:
                         d["output"][k] = d_calc_final["output"]["outcar"][k]
 
-            d["state"] = "successful" if d_calc["has_vasp_completed"] else "unsuccessful"
+            # store optical data
```
Comment (jmmshn) [inline atomate/vasp/drones.py:420]: write the optical data into the task doc

**Comment by utf** [comment]
Thanks! This is really useful.

The code looks good. My only comment is that a lot of those VASP files won't be needed for tests to pass. Would you mind pruning them (especially the large ones) to keep the size of the PR down.

**Comment by jmmshn** [comment]
OK, the files have been pruned

**Comment by utf** [comment]
Perfect, thanks!

**Pull request merged**